ec/ecresource/deploymentresource/schema_apm.go

```diff
@@ -109,39 +109,13 @@ func apmConfig() *schema.Schema {
 		Description:      `Optionally define the Apm configuration options for the APM Server`,
 		Elem: &schema.Resource{
 			Schema: map[string]*schema.Schema{
-				"docker_image": {
-					Type:        schema.TypeString,
-					Description: `A docker URI that allows overriding of the default docker image specified for this version`,
-					Optional:    true,
-				},
-
 				// APM System Settings
 				"debug_enabled": {
 					Type:        schema.TypeBool,
 					Description: `Optionally enable debug mode for APM servers - defaults to false`,
 					Optional:    true,
 					Default:     false,
 				},
-				"elasticsearch_password": {
-					Type:        schema.TypeString,
-					Description: `Optionally override the account within APM - defaults to a system account that always exists (if specified, the username must also be specified)`,
-					Optional:    true,
-				},
-				"elasticsearch_url": {
-					Type:        schema.TypeString,
-					Description: `Optionally override the URL to which to send data (for advanced users only, if unspecified the system selects an internal URL)`,
-					Optional:    true,
-				},
-				"elasticsearch_username": {
-					Type:        schema.TypeString,
-					Description: `Optionally override the account within APM - defaults to a system account that always exists (if specified, the password must also be specified)`,
-					Optional:    true,
-				},
-				"kibana_url": {
-					Type:        schema.TypeString,
-					Description: `Optionally override the URL to which to send data (for advanced users only, if unspecified the system selects an internal URL)`,
-					Optional:    true,
-				},
```

Comment:
What happened to all of these fields? I assume the idea is to parse everything from raw JSON/YAML? just double checking

Reply:
These fields shouldn't be settable by API users and have the potential to cause breakage in the deployment, so removing them because they're not a good thing to have :D. I talked to Stack and Solutions about it and think that's the best course of action too.